src/base/memory_monitor_sr.cpp

```diff
@@ -358,25 +446,82 @@ namespace cubperf
     return 0;
   }
 
-  int memory_monitor::add_stat (THREAD_ENTRY *thread_p, MMON_STAT_ID stat_id, uint64_t size, bool expand)
+  void memory_monitor::add_stat (THREAD_ENTRY *thread_p, MMON_STAT_ID stat_id, uint64_t size)
   {
-    return 0;
+    LOG_TDES *tdes;
+    TRANID trid = NULL_TRANID;
+    int module_idx = MMON_GET_MODULE_IDX_FROM_STAT_ID (stat_id);
+    int comp_map_idx = MMON_GET_COMP_MAP_IDX_FROM_STAT_ID (stat_id);
+    int idx_map_val, comp_idx, subcomp_idx;
+    bool allocation_at_init = !LOG_ISRESTARTED ();
+
+    idx_map_val = m_module[module_idx]->get_comp_idx_map_val (comp_map_idx);
+    comp_idx = get_comp_idx_from_comp_idx_map (idx_map_val);
+    subcomp_idx = get_subcomp_idx_from_comp_idx_map (idx_map_val);
```

Comment:
We need to leave a comment about thess bit operations. Maybe, one month later, we got confused :). Here or somewhere, maybe around the bit operation macros.
like:
```
// purpose : ...
// 32bit MMON_STAT_ID: [module_idx:16 | com_idx_map_idx:16],
// m_comp_idx_map[com_idx_map_idx] -> [comp_idx:16 | subcomp_idx:16 ].
```

Do I get right? I already get a little confused.. lol.